### DISCUSSION TITLE
Treesitter parser for Markdoc

### DISCUSSION THREAD

**Discussion by mohitsinghs** [markdoc/markdoc#337] [Show and tell]
A while back, I created [vscode-markdoc](https://github.com/markdoc-extra/vscode-markdoc) and was trying to implement [a language server](https://github.com/markdoc-extra/markdoc-ls).

Since I was missing syntax highlighting in neovim, I've created [tree-sitter-markdoc](https://github.com/markdoc-extra/tree-sitter-markdoc). This allows syntax highlighting for Markdoc in editors supporting [tree-sitter](https://tree-sitter.github.io/tree-sitter) which includes my favorite neovim. I've ported [grammar tests](https://github.com/markdoc/markdoc/blob/main/src/grammar/tag.test.ts) from Markdoc repo for it.

Feel free to use and improve [these](https://github.com/markdoc-extra) but be warned that these are highly experimental. There is [an existing thread](https://github.com/markdoc/markdoc/discussions/5#discussioncomment-2824702) with screenshots.

**Comment by matv-stripe** [comment]
This is awesome!

**Comment by mfix-stripe** [comment]
Official Markdoc language server out today: https://github.com/markdoc/markdoc/discussions/407!
